layer/decorators/dataset_decorator.py

```diff
@@ -142,27 +142,47 @@ def __init__(self, wrapped: Any, wrapper: Any, enabled: Any) -> None:
         # See https://layerco.slack.com/archives/C02R5B3R3GU/p1646144705414089 for detail.
         def __call__(self, *args: Any, **kwargs: Any) -> Any:
             self.layer.validate()
-            current_project_full_name_ = get_current_project_full_name()
-            config: Config = asyncio_run_in_thread(ConfigManager().refresh())
-            with LayerClient(config.client, logger).init() as client:
-                progress_tracker = RunProgressTracker(
-                    url=config.url,
-                    project_name=current_project_full_name_.project_name,
-                    account_name=current_project_full_name_.account_name,
+            dataset_definition = self.get_definition()
+            dataset_definition.package()
+            if is_feature_active("TAR_PACKAGING"):
+                import subprocess  # nosec: import_subprocess
+                import sys
+
+                subprocess.run(  # nosec: start_process_with_partial_path, subprocess_without_shell_equals_true
+                    [
+                        "sh",
```

Comment:
What about Windows users?

Reply:
With the current form we are not supporting native Windows. We will look into how this plays with WSL.